tests/bindings/test_statevector_nb.py

```diff
@@ -0,0 +1,364 @@
+        state_data[0] = 1.0
+
+        # Initialize with number of qubits first
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Test size
+        assert sv.size() == 2**num_qubits
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_statevector_gate_operations(
+        self, backend_name, precision, nanobind_module, get_statevector_class
+    ):
+        """Test gate operations on StateVectorC64/128 classes."""
+        module = nanobind_module(backend_name)
+
+        StateVectorClass = get_statevector_class(module, precision)
+
+        num_qubits = 2
+        # Create a numpy array representing |0> state with appropriate size
+        dtype = np.complex128 if precision == "128" else np.complex64
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[0] = 1.0
+
+        # Initialize with number of qubits first
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Apply various gates - we can't check the state directly yet,
+        # but we can verify the operations don't raise exceptions
+        sv.PauliX([0], False, [])
+        sv.Hadamard([0], True, [])
+        sv.Hadamard([1], False, [])
+        sv.CNOT([0, 1], True, [])
+
+        # If we get here without exceptions, the test passes
+        assert True
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_statevector_parametric_gates(
+        self, backend_name, precision, nanobind_module, get_statevector_class
+    ):
+        """Test parametric gates on StateVectorC64/128 classes."""
+        module = nanobind_module(backend_name)
+
+        StateVectorClass = get_statevector_class(module, precision)
+
+        num_qubits = 1
+        # Create a numpy array representing |0> state with appropriate size
+        dtype = np.complex128 if precision == "128" else np.complex64
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[0] = 1.0
+
+        # Initialize with number of qubits first
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Apply parametric gates - we can't check the state directly yet,
+        # but we can verify the operations don't raise exceptions
+        sv.RX([0], False, [np.pi / 2])
+        sv.RY([0], False, [np.pi])
+        sv.RZ([0], False, [np.pi / 2])
+
+        # If we get here without exceptions, the test passes
+        assert True
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_statevector_matrix_application(
+        self, backend_name, precision, nanobind_module, get_statevector_class
+    ):
+        """Test matrix application on StateVectorC64/128 classes."""
+        # Skip non-qubit backends for this test
+        if backend_name != "qubit":
+            pytest.skip(f"Matrix application test only supported for qubit backend")
```

Comment:
Doesn't this check already happen at the module level?

Reply:
removed